server/src/utils/secureLinkPreview.ts

```
@@ -0,0 +1,63 @@
+function handleRedirects(url: string, newUrl: string, maxRedirects: number): Promise<boolean> {
+  return new Promise(async (resolve, reject) => {
+    if (maxRedirects < 0) {
+      return reject(new Error("Exceeded max redirects."));
+    }
+    try {
+      await resolveDns(newUrl); // Validate the new URL's IP
+      resolve(true); // Allow the redirect
+    } catch (error) {
+      reject(error); // Reject if the redirect target is disallowed
+    }
+  });
```

Biome (2.1.2)

[error] Promise executor functions should not be `async`.



(lint/suspicious/noAsyncPromiseExecutor)